storage/src/mmr/mem.rs

```diff
@@ -247,6 +247,37 @@ impl<H: CHasher> Mmr<H> {
         self.oldest_retained_pos = pos;
     }
 
+    // A lightweight cloning operation that "clones" only the fully pruned state of this MMR. The output is exactly the
+    // same as the result of mmr.prune_all(), only you get a copy without mutating the original.
+    //
+    // Overhead is Log_2(n) in the number of elements even if the original MMR is fully unpruned.
+    pub fn clone_pruned(&self) -> Self {
```

Comment:
Would be great to create an issue to "collapse" a cloned tree into a "root" tree (imagine that could be useful near the tip of a blockchain).

Reply:
Yeah will do, I like that idea.